migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/table.rs

```diff
@@ -39,15 +40,15 @@ impl<'schema> TableDiffer<'schema> {
         })
     }
 
-    pub(crate) fn created_foreign_keys(&self) -> impl Iterator<Item = &ForeignKey> {
+    pub(crate) fn created_foreign_keys<'a>(&'a self) -> impl Iterator<Item = ForeignKeyRef<'a, 'schema>> + 'a {
```

Comment:
I might be wrong, but isn't it possible now to write this without the `'a` lifetime? Not entirely sure if this is better tho, but feel it makes the signature a bit easier to read?

```suggestion
    pub(crate) fn created_foreign_keys(&self) -> impl Iterator<Item = ForeignKeyRef<'_, 'schema>> {
```

Reply:
Possibly, I tend to add the lifetimes as a defensive measure, but this could work (depending on whether it can infer that the `impl Iterator` has the implicit lifetime.